clients/src/main/java/org/apache/kafka/common/record/DefaultRecord.java

```diff
@@ -266,7 +272,7 @@ private static long computeChecksum(long timestamp,
 
     @Override
     public boolean hasMagic(byte magic) {
-        return magic >= MAGIC_VALUE_V2;
+        return magic == MAGIC_VALUE_V2;
```

Comment:
This is admittedly an awkward api since there is no record-level magic value in the new format. Maybe it should be `isCompatibleWithMagic`? In the future, if we add new fields or remove existing ones, we can use the presence of those fields to indicate whether or not the record is compatible.